pkg/modules/promote/implpromote/handler.go

```diff
@@ -0,0 +1,141 @@
+package implpromote
+
+import (
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	schemamigrator "github.com/SigNoz/signoz-otel-collector/cmd/signozschemamigrator/schema_migrator"
+	"github.com/SigNoz/signoz/pkg/errors"
+	"github.com/SigNoz/signoz/pkg/http/render"
+	"github.com/SigNoz/signoz/pkg/modules/promote"
+	"github.com/SigNoz/signoz/pkg/telemetrylogs"
+	"github.com/SigNoz/signoz/pkg/types/authtypes"
+	"github.com/SigNoz/signoz/pkg/types/promotetypes"
+)
+
+type handler struct {
+	module promote.Module
+}
+
+func NewHandler(module promote.Module) promote.Handler {
+	return &handler{module: module}
+}
+
+func (h *handler) HandlePromote(w http.ResponseWriter, r *http.Request) {
+	_, err := authtypes.ClaimsFromContext(r.Context())
+	if err != nil {
+		render.Error(w, errors.NewInternalf(errors.CodeInternal, "failed to get org id from context"))
+		return
+	}
+
+	switch r.Method {
+	case http.MethodGet:
+		h.GetPromotedAndIndexedPaths(w, r)
+		return
+	case http.MethodPost:
+		h.PromotePaths(w, r)
+		return
+	case http.MethodDelete:
+		h.DropIndex(w, r)
+		return
+	default:
+		render.Error(w, errors.NewMethodNotAllowedf(errors.CodeMethodNotAllowed, "method not allowed"))
+		return
+	}
+}
+
+func (h *handler) DropIndex(w http.ResponseWriter, r *http.Request) {
+	var req promotetypes.PromotePath
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		render.Error(w, errors.NewInvalidInputf(errors.CodeInvalidInput, "Invalid data"))
+		return
```

Comment:
nits: 

1. Please use `binding.JSON.BindBody` in http handlers. It helps with validation and field type error messages. We will add this to the contributing guidelines, i.e., use `binding.JSON.BindBody` for reading request payloads.
2. Override the json marshal and unmarshal and return the validation error from there, and return the error up the call stack. Example https://github.com/SigNoz/signoz/blob/e66bfe59618917ed947135ba73bb00086e47a1e4/pkg/types/parsertypes/queryfilteranalyzer.go#L18 What do you think?

Reply:
used binding.JSON.BindBody